projects/default/controllers/ros/RosCamera.cpp

```diff
@@ -107,7 +113,41 @@ void RosCamera::publishValue(ros::Publisher publisher) {
   publisher.publish(image);
 }
 
+sensor_msgs::CameraInfo RosCamera::createCameraInfoMessage() {
+  sensor_msgs::CameraInfo info;
+  info.header.stamp = ros::Time::now();
+  info.header.frame_id = mFrameIdPrefix + RosDevice::fixedDeviceName();
+
+  double width = mCamera->getWidth();
+  double height = mCamera->getHeight();
+  info.width = mCamera->getWidth();
+  info.height = mCamera->getHeight();
+
+  double hfov = mCamera->getFov();
+  double focalLength = width / (2.0 * tan(hfov / 2.0));
+
+  double fx, fy, cx, cy;
+  fx = focalLength;
+  fy = focalLength;
+  cx = (width + 1.0) / 2.0;
+  cy = (height + 1.0) / 2.0;
+  boost::array<double, 9> K = {fx, 0.0, cx, 0.0, fy, cy, 0.0, 0.0, 1.0};
+  info.K = K;
+
+  boost::array<double, 9> R = {1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0};
+  info.R = R;
+
+  boost::array<double, 12> P = {fx, 0.0, cx, 0.0, 0.0, fy, cy, 0.0, 0.0, 0.0, 1.0, 0.0};
+  info.P = P;
+
+  return info;
+}
+
 void RosCamera::publishAuxiliaryValue() {
+  if (mCameraInfoPublisher.getNumSubscribers() > 0) {
+    mCameraInfoPublisher.publish(createCameraInfoMessage());
+  }
```

Comment:
Coding style [#CS360](https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style#cs360-single-if-while-or-for-statement-must-be-written-without-brackets-the-statement-part-should-be-put-on-a-separate-line-in-case-of-nested-block-keep-brackets):
```suggestion
  if (mCameraInfoPublisher.getNumSubscribers() > 0)
    mCameraInfoPublisher.publish(createCameraInfoMessage());
```